src/main/java/com/payline/payment/sandbox/service/impl/NotificationServiceImpl.java

```diff
@@ -24,77 +25,81 @@ public NotificationResponse parse(NotificationRequest notificationRequest) {
-                        .withTransactionStatus(
-                                OnHoldTransactionStatus.builder()
-                                        .onHoldCause(OnHoldCause.SCORING_ASYNC)
-                                        .build()
-                        )
-                        .build();
+                /* TransactionStateChangedResponse */
+                case "40200":
+                    Logger.log(this.getClass().getSimpleName(), "notificationRequest", amount, "TransactionStateChangedResponse");
+                    return TransactionStateChangedResponse.TransactionStateChangedResponseBuilder.aTransactionStateChangedResponse()
+                            .build();
+                case "40201":
+                    Logger.log(this.getClass().getSimpleName(), "notificationRequest", amount, "TransactionStateChangedResponse avec SuccessTransactionStatus");
+                    return TransactionStateChangedResponse.TransactionStateChangedResponseBuilder.aTransactionStateChangedResponse()
+                            .withTransactionStatus(SuccessTransactionStatus.builder().build())
+                            .build();
+                case "40202":
+                    Logger.log(this.getClass().getSimpleName(), "notificationRequest", amount, "TransactionStateChangedResponse avec FailureTransactionStatus");
+                    return TransactionStateChangedResponse.TransactionStateChangedResponseBuilder.aTransactionStateChangedResponse()
+                            .withTransactionStatus(
+                                    FailureTransactionStatus.builder()
+                                            .failureCause(FailureCause.INVALID_DATA)
+                                            .build()
+                            )
+                            .build();
+                case "40203":
+                    Logger.log(this.getClass().getSimpleName(), "notificationRequest", amount, "TransactionStateChangedResponse avec OnHoldTransactionStatus");
+                    return TransactionStateChangedResponse.TransactionStateChangedResponseBuilder.aTransactionStateChangedResponse()
+                            .withTransactionStatus(
+                                    OnHoldTransactionStatus.builder()
+                                            .onHoldCause(OnHoldCause.SCORING_ASYNC)
+                                            .build()
+                            )
+                            .build();
 
-            /* Generic plugin errors */
-            default:
-                return super.generic(this.getClass().getSimpleName(),"notificationRequest", amount );
+                /* Generic plugin errors */
+                default:
+                    return super.generic(this.getClass().getSimpleName(), "notificationRequest", amount);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+            return null;
```

Comment:
@JanFLORENT 
Faut-il faire quelque chose de particulier pour l'exception ? 
Par exemple `throw new RuntimeException("Plugin error: unable to load ressource", e);`